torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -210,6 +212,26 @@ def _check_flat_params_on_expected_device(state: _FSDPState, module: nn.Module):
             )
 
 
+def _init_device_mesh(
+    root_state: _FSDPState,
+) -> Optional[DeviceMesh]:
+    # We are testing 1D DeviceMesh where dist.get_world_size(pg) == dist.get_world_size() for now.
+    # TODO: Address cases when dist.get_world_size(pg) != dist.get_world_size(). This would capture
+    #       what 1D DeviceMesh currently would not work for:
+    #       1) HSDP Hybrid Sharding, 2) 2D FSDP + TP, 3) dist.new_group() cannot be expressed in 1D DeviceMesh.
+    if get_world_size(root_state.process_group) != get_world_size():
+        return None
+    # Temporarily skip DeviceMesh initialization when the backend is set to fake.
+    # TODO: Remove the condition once the issue of "Default PG backend: fake not supporting CUDA!" is resolved.
+    if get_backend() == "fake" or not root_state.compute_device:
+        return None
+    device_type = root_state.compute_device.type
+    mesh_tensor = torch.arange(get_world_size(root_state.process_group))
+    device_mesh = DeviceMesh(device_type, mesh_tensor, _init_process_groups=False)
+    device_mesh._dim_groups = [root_state.process_group]
```

Comment:
We should fix the PG<>DM interop as part of this effort. This piece of code is not particularly great.
CC @wanchaol